cas-server-integration-ehcache/src/main/java/org/jasig/cas/ticket/registry/EhCacheTicketRegistry.java

```diff
@@ -50,7 +51,7 @@
  * @author Andrew Tillinghast
  * @since 3.5
  */
-public final class EhCacheTicketRegistry extends AbstractDistributedTicketRegistry implements InitializingBean {
+public final class EhCacheTicketRegistry extends AbstractCrypticTicketRegistry implements InitializingBean {

```

Comment:
The `ticketRegistry` is defined in the `ticketRegistry.xml` file, so that's the only place you need to change something: rename the default one and create this new one. I don't think it's too much work, not more than setting the cypher to use.
And yes, we would need to update the documentation, but this is something we cannot avoid, even with your proposal.

I'm not sure to understand why some ticket registries would not support hashing / encryption?